modules/search/class.jetpack-search-widget-filters.php

```diff
@@ -98,22 +107,157 @@ function update( $new_instance, $old_instance ) {
+					<?php esc_html_e( 'Title:', 'jetpack' ); ?>
+				</label>
+				<input
+					class="widefat"
+					id="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>"
+					name="<?php echo esc_attr( $this->get_field_name( 'title' ) ); ?>"
+					type="text"
+					value="<?php echo esc_attr( $title ); ?>"
+				/>
+			</p>
+			<?php foreach ( (array) $instance['filters'] as $filter ) : ?>
+				<?php $this->render_widget_filter( $filter ); ?>
+			<?php endforeach; ?>
+		</div>
 		<?php
 	}
 
+	function render_widget_filter( $filter ) {
+		$args = wp_parse_args( $filter, array(
+			'name' => '',
+			'type' => 'taxonomy',
+			'taxonomy' => '',
+			'post_type' => '',
+			'count' => 10,
+		) );
+
+		?>
+		<div class="<?php echo sprintf( 'jetpack-search-filters-widget__filter is-%s', sanitize_key( $args['type'] ) ); ?>">
+			<p>
+				<label>
+					<?php esc_html_e( 'Filter Name:', 'jetpack' ); ?>
+					<input
+						class="widefat"
+						type="text"
+						name="<?php echo esc_attr( $this->get_field_name( 'filter_name' ) ); ?>[]"
+						value="<?php
+							echo ! empty( $args['name'] )
+								? esc_attr( $args['name'] )
+								: '';
+						?>"
+					/>
+				</label>
+			</p>
+
+			<p>
+				<label>
+					<?php esc_html_e( 'Filter Type:', 'jetpack' ); ?>
+					<select name="<?php echo esc_attr( $this->get_field_name( 'filter_type' ) ); ?>[]" class="widefat filter-select">
+						<option value="taxonomy" <?php selected( $args['type'], 'taxonomy' ); ?>>
+							<?php esc_html_e( 'Taxonomy', 'jetpack' ); ?>
+						</option>
+						<option value="post_type" <?php selected( $args['type'], 'post_type' ); ?>>
+							<?php esc_html_e( 'Post Type', 'jetpack' ); ?>
+						</option>
+						<option value="date_histogram" <?php selected( $args['type'], 'date_histogram' ); ?>>
+							<?php esc_html_e( 'Date', 'jetpack' ); ?>
+						</option>
+					</select>
+				</label>
+			</p>
+
+			<p class="jetpack-search-filters-widget__taxonomy-select">
+				<label>
+					<?php esc_html_e( 'Choose a taxonomy:', 'jetpack' ); ?>
+					<select name="<?php echo esc_attr( $this->get_field_name( 'taxonomy_type' ) ); ?>[]" class="widefat">
+						<?php foreach ( get_taxonomies( false, 'objects' ) as $taxonomy ) : ?>
+							<option value="<?php echo esc_attr( $taxonomy->name ); ?>" <?php selected( $taxonomy->name, $args['taxonomy'] ); ?>>
+								<?php echo esc_html( $taxonomy->label ); ?>
+							</option>
+						<?php endforeach; ?>
+					</select>
+				</select>
+				</label>
+			</p>
+
+			<p class="jetpack-search-filters-widget__post-type-select">
+				<label>
+					<?php esc_html_e( 'Choose a post type:', 'jetpack' ); ?>
+					<select name="<?php echo esc_attr( $this->get_field_name( 'post_type' ) ); ?>[]" class="widefat">
+						<?php foreach ( get_post_types( false, 'objects' ) as $post_type ) : ?>
+							<option value="<?php echo esc_attr( $post_type->name ); ?>" <?php selected( $post_type->name, $args['post_type'] ); ?>>
```

Comment:
Don't need a list of post types here. When the user selects post type, that means we just run an aggregation on the post type, we don't need an extra input to select which type.

https://cloudup.com/cepkzisz4OX